zcash_client_sqlite/src/wallet/sapling.rs

```diff
@@ -1451,63 +1162,47 @@ pub(crate) mod tests {
     #[test]
     #[cfg(feature = "transparent-inputs")]
     fn shield_transparent() {
-        let cache_file = NamedTempFile::new().unwrap();
-        let db_cache = BlockDb(Connection::open(cache_file.path()).unwrap());
-        init_cache_database(&db_cache).unwrap();
-
-        let data_file = NamedTempFile::new().unwrap();
-        let mut db_data = WalletDb::for_path(data_file.path(), testing::network()).unwrap();
-        init_wallet_db(&mut db_data, None).unwrap();
+        let mut test = TestBuilder::new().with_block_cache().build();
 
         // Add an account to the wallet
         let seed = Secret::new([0u8; 32].to_vec());
-        let (account_id, usk) = db_data.create_account(&seed).unwrap();
+        let (account_id, usk) = test.wallet_mut().create_account(&seed).unwrap();
         let dfvk = usk.sapling().to_diversifiable_full_viewing_key();
-        let uaddr = db_data.get_current_address(account_id).unwrap().unwrap();
+        let uaddr = test
+            .wallet()
+            .get_current_address(account_id)
+            .unwrap()
+            .unwrap();
         let taddr = uaddr.transparent().unwrap();
 
+        // Ensure that the wallet has at least one block
```

Comment:
~This change in behavior of the test seems innocuous, but the comment should elaborate a little on why it's necessary.~ realized later that this is a move.

Reply:
Yeah, AFAICT the test was not relying on the ordering here; it was just pretending that the UTXO was received in the fake block without actually doing so. This makes the height of the latter accessible to the former.